### ISSUE TITLE
Market.FROM_TOKEN causes HTTP 400

### ISSUE THREAD

**Issue by chuck-stein** [adamint/spotify-web-api-kotlin#324]
**Describe the bug**
When making a request with `Market.FROM_TOKEN`, the request will fail with `SpotifyException.BadRequestException` due to an HTTP 400 error.

**To Reproduce**
1. Call an endpoint that takes in a market parameter, specifying `Market.FROM_TOKEN`. For example:

```kotlin
spotifyClientApi.browse.getRecommendations(market = Market.FROM_TOKEN, seedTracks = listOf(/* any track IDs */))
```
2. `SpotifyException.BadRequestException` is thrown, with error code 400.

**Expected behavior**
A successful response from the endpoint, providing content (such as recommended tracks) relevant to the user's current market/country (pulled from their auth token).

**Smartphone (please complete the following information):**
 - Device: OnePlus 7 Pro
 - OS: Android 11
 - Browser: Native Android app
 - Version: library version 3.8.8

**Additional context**
The 400 error is happening because the `market` parameter is resolved by getting the enum name, which is `FROM_TOKEN`. However, this is not a valid Spotify market -- it should be `from_token` in all lowercase instead.

**Comment by adamint** [comment]
Yep, this is correct. Market.name is being used instead of Market.marketName, I will update.

**Comment by adamint** [comment]
Closing with release 4.0.1. Please download and try!

**Comment by chuck-stein** [comment]
Thanks for looking into this. Unfortunately it looks like the fix only affected certain endpoints. After updating to the latest version, I am still getting 400s for the endpoints I was hitting, namely:

- `browse.getRecommendations`
- `browse.getCategoryList`
- `albums.getAlbum`
- `artists.getArtistAlbums`

I do know that I updated correctly, because if I click into the `artists.getArtistTopTracks` endpoint, for example, I do see it using the new `market.getSpotifyId()` extension. However, if I click into the endpoints I listed above, I still see them using `market?.name`.

I'm also guessing that isn't an exhaustive list, so it's probably worth just searching for and replacing all usages of `market.name`. Thanks again for the help!
